pkg/build/controller/strategy/util.go

```diff
@@ -269,19 +270,35 @@ func getPodLabels(build *buildapi.Build) map[string]string {
 	return map[string]string{buildapi.BuildLabel: buildapi.LabelValue(build.Name)}
 }
 
-func setOwnerReference(pod *v1.Pod, build *buildapi.Build) {
+func makeOwnerReference(build *buildapi.Build) metav1.OwnerReference {
 	t := true
-	pod.OwnerReferences = []metav1.OwnerReference{
-		{
-			APIVersion: BuildControllerRefKind.GroupVersion().String(),
-			Kind:       BuildControllerRefKind.Kind,
-			Name:       build.Name,
-			UID:        build.UID,
-			Controller: &t,
-		},
+	return metav1.OwnerReference{
+		APIVersion: BuildControllerRefKind.GroupVersion().String(),
+		Kind:       BuildControllerRefKind.Kind,
+		Name:       build.Name,
+		UID:        build.UID,
+		Controller: &t,
 	}
 }
 
+func setOwnerReference(pod *v1.Pod, build *buildapi.Build) {
+	pod.OwnerReferences = []metav1.OwnerReference{makeOwnerReference(build)}
+}
+
+// HasOwnerReference returns true if the build pod has an OwnerReference to the
+// build.
+func HasOwnerReference(pod *v1.Pod, build *buildapi.Build) bool {
```

Comment:
How about renaming the method to `HaveACommonOwnerReference`

Reply:
it's not so much a common owner ref.  it's one owning the other.  I think the name is fine but if we want something else, "Owns()" would be the name i would suggest.  Or "BuildOwnsPod" is perhaps clearest about the relationship being tested.